src/pages/patch/ParametersModal.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import { Modal } from "components/Modal";
+import { Parameter } from "types/patch";
+
+interface ParametersProps {
+  parameters: Parameter[];
+  dataCy: string;
+  closeModal: () => void;
+  visible: boolean;
+}
+export const ParametersModal: React.FC<ParametersProps> = ({
+  parameters,
+  dataCy,
+  closeModal,
+  visible,
+}) => (
+  <Modal
+    visible={visible}
+    footer={null}
+    onCancel={closeModal}
+    data-cy={dataCy}
+    title="Patch Parameters"
+  >
+    <span>
+      {parameters?.map(
+        (param) =>
+          param && (
+            <StyledBadge key={`param_${param?.key}`}>
+              {param?.key}:{param?.value}
```

Comment:
is there a scenario where `parameters` will exist, as will `param`, but the `key` and `value` fields are null or undefined? i ask because this code indicates that a StyledBadge can be rendered with an undefined, i.e. blank, `key` and `value`

Reply:
key has to be defined. value can be the empty string, but it's still turned into a pointer so I suppose it also can't be undefined. I assumed we used ?. for everything just to be safe but I can change them both if that's preferable